internal/inventory/azurefetcher/fetcher_resource_graph.go

```diff
@@ -0,0 +1,103 @@
+	return &resourceGraphFetcher{
+		logger:   logger,
+		provider: provider,
+	}
+}
+
+func (f *resourceGraphFetcher) Fetch(ctx context.Context, assetChan chan<- inventory.AssetEvent) {
+	resourcesToFetch := []struct {
+		name           string
+		azureGroup     string
+		azureType      string
+		classification inventory.AssetClassification
+	}{
+		{"App Services", azurelib.AssetGroupResources, azurelib.WebsitesAssetType, inventory.AssetClassificationAzureAppService},
+		{"Container Registries", azurelib.AssetGroupResources, azurelib.ContainerRegistryAssetType, inventory.AssetClassificationAzureContainerRegistry},
+		{"Cosmos DB Accounts", azurelib.AssetGroupResources, azurelib.DocumentDBDatabaseAccountAssetType, inventory.AssetClassificationAzureCosmosDBAccount},
+		{"Cosmos DB SQL Databases", azurelib.AssetGroupResources, azurelib.CosmosDBForSQLDatabaseAssetType, inventory.AssetClassificationAzureCosmosDBSQLDatabase},
+		{"Disks", azurelib.AssetGroupResources, azurelib.DiskAssetType, inventory.AssetClassificationAzureDisk},
+		{"Elastic Pools", azurelib.AssetGroupResources, azurelib.ElasticPoolAssetType, inventory.AssetClassificationAzureElasticPool},
+		{"MySQL Flexible Servers", azurelib.AssetGroupResources, azurelib.FlexibleMySQLDBAssetType, inventory.AssetClassificationAzureSQLServer},
+		{"Resource Groups", azurelib.AssetGroupResourceContainers, azurelib.ResouceGroupAssetType, inventory.AssetClassificationAzureResourceGroup},
+		{"SQL Databases", azurelib.AssetGroupResources, azurelib.MySQLDatabaseAssetType, inventory.AssetClassificationAzureSQLDatabase},
+		{"Snapshots", azurelib.AssetGroupResources, azurelib.SnapshotAssetType, inventory.AssetClassificationAzureSnapshot},
+		{"Storage Accounts", azurelib.AssetGroupResources, azurelib.StorageAccountAssetType, inventory.AssetClassificationAzureStorageAccount},
+		{"Virtual Machines", azurelib.AssetGroupResources, azurelib.VirtualMachineAssetType, inventory.AssetClassificationAzureVirtualMachine},
+	}
+	for _, r := range resourcesToFetch {
+		f.fetch(ctx, r.name, r.azureGroup, r.azureType, r.classification, assetChan)
+	}
+}
+
+func (f *resourceGraphFetcher) fetch(ctx context.Context, resourceName, resourceGroup, resourceType string, classification inventory.AssetClassification, assetChan chan<- inventory.AssetEvent) {
+	f.logger.Infof("Fetching %s", resourceName)
+	defer f.logger.Infof("Fetching %s - Finished", resourceName)
+
+	azureAssets, err := f.provider.ListAllAssetTypesByName(ctx, resourceGroup, []string{resourceType})
+	if err != nil {
+		f.logger.Errorf("Could not fetch %s: %v", resourceName, err)
+		return
+	}
+
+	for _, item := range azureAssets {
+		name := item.Name
+		if name == "" {
+			name = item.DisplayName
+		}
+		assetChan <- inventory.NewAssetEvent(
+			classification,
+			[]string{item.Id},
+			name,
+			inventory.WithRawAsset(item),
+			inventory.WithCloud(inventory.AssetCloud{
```

Comment:
can / should we get `Region`, `Project` too?

Reply:
Is it sane to assume `Project == ResourceGroup` in this case?

Reply:
i was thinking `Subscription` might me more suitable, but both are assumptions. so maybe leaving empty is actually better